### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by tudorp9k**
Backend authentication with Microsoft Identity.

### CODE REVIEW

**Comment by Paul-Podoleanu** [comment]
Aprob :3

File: back-end/YummyGen/YummyGen.Application/AuthenticationService.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public class AuthenticationService : IAuthenticationService
+    {
+        private readonly UserManager<User> userManager;
+
+        public AuthenticationService(UserManager<User> userManager)
+        {
+            this.userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+        }
+
+        public async Task<UserDto> Register(RegisterDto userDto)
```
Comment (roommate6) [inline back-end/YummyGen/YummyGen.Application/AuthenticationService.cs]: rename userDto with registerDto. For the loginDto you've done that, but for the register, not...

File: back-end/YummyGen/YummyGenAPI/Startup.cs [hunk 2]
```diff
@@ -17,13 +19,25 @@ public static void RegisterServices(WebApplicationBuilder builder)
             var connectionString = builder.Configuration.GetConnectionString("WebApiDatabase");
             builder.Services.AddDbContext<ApplicationDbContext>(options => options.UseNpgsql(connectionString));
 
+            builder.Services.AddIdentity<User, IdentityRole>(options =>
```
Comment (roommate6) [inline back-end/YummyGen/YummyGenAPI/Startup.cs:22]: Good job. I never knew that you can use such a powerful tool for passwords. I will consider now using this in feature projects. Ty

**Review by roommate6 — CHANGES_REQUESTED** [review]
Tested the register api. It works. Good job! But before merging with the main, check the feedback on some lines.

File: back-end/YummyGen/YummyGen.Application/AuthenticationService.cs (hunk 1 above)
Comment (tudorp9k) [reply back-end/YummyGen/YummyGen.Application/AuthenticationService.cs]: Updated 👍

**Review by roommate6 — APPROVED** [review]
now everything is good

**Pull request merged**